src/main/java/org/sonar/ucfg/UCFGElement.java

```diff
@@ -24,12 +24,12 @@
 import java.util.stream.Collectors;
 import javax.annotation.Nullable;
 
-public abstract class Instruction {
+public abstract class UCFGElement {
 
   private final InstructionType type;
   final LocationInFile locationInFile;
 
-  private Instruction(InstructionType type, @Nullable LocationInFile locationInFile) {
+  private UCFGElement(InstructionType type, @Nullable LocationInFile locationInFile) {
```

Comment:
I'm not sure that `InstructionType` is the correct naming now. What about `UcfgElementType`?